python/lsst/sims/catUtils/utils/LightCurveGenerator.py

```diff
@@ -0,0 +1,496 @@
+from __future__ import print_function
+import numpy as np
+import copy
+
+from lsst.sims.catUtils.utils import ObservationMetaDataGenerator
+from lsst.sims.catUtils.mixins import PhotometryStars, VariabilityStars
+from lsst.sims.catUtils.mixins import PhotometryGalaxies, VariabilityGalaxies
+from lsst.sims.catalogs.measures.instance import InstanceCatalog, compound, cached
+from lsst.sims.utils import haversine
+
+import time
+
+__all__ = ["StellarLightCurveGenerator", "AgnLightCurveGenerator",
+           "_baseLightCurveCatalog", "LightCurveGenerator"]
+
+# a global cache to store SedLists loaded by the light curve catalogs
+_sed_cache = {}
+
+
+class _baseLightCurveCatalog(InstanceCatalog):
+    """
+    """
+
+    column_outputs = ["uniqueId", "raJ2000", "decJ2000",
+                      "lightCurveMag", "sigma_lightCurveMag",
+                      "truthInfo"]
+
+    def iter_catalog(self, chunk_size=None, query_cache=None):
+        """
+        Returns an iterator over rows of the catalog.
+
+        Parameters
+        ----------
+        chunk_size : int, optional, defaults to None
+            the number of rows to return from the database at a time. If None,
+            returns the entire database query in one chunk.
+
+        query_cache : iterator over database rows, optional, defaults to None
+            the result of calling db_obj.query_columns().  If query_cache is not
+            None, this method will iterate over the rows in query_cache and produce
+            an appropriate InstanceCatalog. DO NOT use this
+            unless you know what you are doing.  It is an optional
```

Comment:
- Is it useful to use the _baseLightCurve and not use query_cache? Maybe the use of custom LightCurveGenerator classes (as opposed to those already provided by catsim) should be discouraged rather than a single variable. 
- Even if the above is not true, DO NOT use should probably be replaced by something saying that non None values should not be used.